src/Bicep.Core.Samples/InvalidExpressions_LF/main.diagnostics.bicep

```diff
@@ -338,9 +339,10 @@ var badInnerArrayIndexer = sampleObject.myInner.otherArr['s']
 var badIndexer = sampleObject.myStr['s']
 //@[17:35) [BCP076 (Error)] Cannot index over expression of type 's'. Arrays or objects are required. |sampleObject.myStr|
 var badInnerArray = sampleObject.myInner.fakeArr['s']
+//@[20:48) [BCP076 (Error)] Cannot index over expression of type error. Arrays or objects are required. |sampleObject.myInner.fakeArr|
```

Comment:
>Error [](start = 20, length = 5)

Why did this error appear?

Reply:
Updated UnassignableType, before I had a condition that checked if errors (ImmutableArray) is null or empty then sets TypeKind to UnassignableType otherwise TypeKind is set to ErrorType, seems like checking for Name == "Errors" (constant value) did the trick, do we have cases where we pass a null or empty errors list?

Reply:
Yeah, there are places where the error type can have an empty set of diagnostics attached.